caffe2/utils/Optional.h

```diff
@@ -179,13 +179,11 @@ constexpr struct in_place_t{} in_place{};
 
```

Comment:
Hi @Yangqing, one simple question here: why we need to keep this version of optional in caffe2/utils, given that we already have a optional.h in c10?